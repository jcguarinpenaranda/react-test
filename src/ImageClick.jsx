import React from 'react';
import PropTypes from 'prop-types';

/**
* Retorna una imagen, a la que al hacÃ©rsele
* click, muestra un alert al usuario
*/
const ImageClick = props => (
  <button onClick={props.onImgClick}>
    <img alt="Paisaje" src="http://www.fondos10.net/wp-content/uploads/2010/04/Hermoso-paisaje.jpg" />
  </button>
);

ImageClick.propTypes = {
  onImgClick: PropTypes.func.isRequired,
};

export default ImageClick;
