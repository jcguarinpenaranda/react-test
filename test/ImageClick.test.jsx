import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ImageClick from '../src/ImageClick';

test('should be defined', () => {
  expect(ImageClick).toBeTruthy();
});

test('should mount correctly', () => {
  const m = mount(<ImageClick onImgClick={() => {}} />);
  expect(m.html()).toBeTruthy();
});

test('button onClick should run my function', () => {
  const fn = jest.fn();
  const m = mount(<ImageClick onImgClick={fn} />);
  const button = m.find('button');
  button.simulate('click');
  // const buttonOnClick = button.prop('onClick');
  // buttonOnClick();

  expect(fn).toHaveBeenCalledTimes(1);
});

test('should have the correct image', () => {
  const m = mount(<ImageClick onImgClick={() => {}} />);
  const img = m.find('img');
  expect(img.prop('src')).toEqual('http://www.fondos10.net/wp-content/uploads/2010/04/Hermoso-paisaje.jpg');
});

test('should match snapshot', () => {
  const m = mount(<ImageClick onImgClick={() => {}} />);
  expect(toJSON(m)).toMatchSnapshot();
});
