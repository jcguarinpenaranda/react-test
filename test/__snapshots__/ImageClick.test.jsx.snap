// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<ImageClick
  onImgClick={[Function]}
>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "purple",
      }
    }
  >
    <img
      alt="Paisaje"
      src="http://www.fondos10.net/wp-content/uploads/2010/04/Hermoso-paisaje.jpg"
    />
  </button>
</ImageClick>
`;
